internal/agent/device/fileio/managed_file.go

```diff
@@ -0,0 +1,132 @@
+package fileio
+
+import (
+	"bytes"
+	"fmt"
+	"os"
+	"path/filepath"
+
+	ign3types "github.com/coreos/ignition/v2/config/v3_4/types"
+)
+
+type ManagedFile interface {
+	Path() string
+	Exists() (bool, error)
+	ShouldSkip() (bool, error)
+	Write() error
+}
+
+type managedFile struct {
+	ign3types.File
+	initialized bool
+	exists      bool
+	size        int64
+	contents    []byte
+	rootDir     string
+}
+
+func newManagedFile(f ign3types.File, rootDir string) ManagedFile {
```

Comment:
Inside managed file?

Reply:
Done

Reply:
What will it be used for?

Reply:
the managed file should be a light layer on top of a writer otherwise you are adding writer implementation to the managedFile which I would like to avoid.

Reply:
I don't see any code that I want to move from managed file to writer, or to use any existing functionality.
Leaving the code as is for now.